compiler-rt/lib/asan/asan_win.cpp

```diff
@@ -166,6 +190,15 @@ INTERCEPTOR_WINAPI(HANDLE, CreateThread, LPSECURITY_ATTRIBUTES security,
                             thr_flags, tid);
 }
 
+INTERCEPTOR_WINAPI(void, ExitThread, DWORD dwExitCode) {
+  DWORD key = atomic_load(&g_native_tls_key, memory_order_relaxed);
+  AsanThread *t = (AsanThread *)TlsGetValue(key);
```

Comment:
Isn't this  the same as in SetCurrentThread ?

Reply:
Different, SetCurrentThread calls AsanTSDSet to save the context

Reply:
Of course there is another problem with the log
```
  CHECK(t->context());
  VReport(2, "SetCurrentThread: %p for thread %p\n", (void *)t->context(),
          (void *)GetThreadSelf());
  // Make sure we do not reset the current AsanThread.
  CHECK_EQ(0, AsanTSDGet());
---
// In contrast to POSIX, on Windows GetCurrentThreadId()
// returns a system-unique identifier.
tid_t GetTid() {
  return GetCurrentThreadId();
}

uptr GetThreadSelf() {
  return GetTid();
}
```
GetCurrentThreadId is returned a handle on Windows

Reply:
> So why we can't just GetCurrentThread()->Destroy()?

Thanks, review it again when you have time